module/metrics/execution.go

```diff
@@ -375,11 +378,34 @@ func NewExecutionCollector(tracer module.Tracer) *ExecutionCollector {
 		Buckets:   []float64{-1, 0, 10_000_000, 100_000_000, 1_000_000_000},
 	})
 
-	totalChunkDataPackRequests := promauto.NewCounter(prometheus.CounterOpts{
+	chunkDataPackRequestProcessedTotal := promauto.NewCounter(prometheus.CounterOpts{
 		Namespace: namespaceExecution,
 		Subsystem: subsystemProvider,
-		Name:      "chunk_data_packs_requested_total",
-		Help:      "the total number of chunk data pack requests provider engine received",
+		Name:      "chunk_data_pack_requested_total",
+		Help:      "the total number of chunk data pack requests processed by provider engine",
+	})
+
+	chunkDataPackResponseDeliveredTotal := promauto.NewCounter(prometheus.CounterOpts{
+		Namespace: namespaceExecution,
+		Subsystem: subsystemProvider,
+		Name:      "chunk_data_pack_delivered_total",
+		Help:      "the total number of chunk data pack responses delivered by provider engine",
+	})
+
+	chunkDataPackResponseDispatchTime := promauto.NewHistogram(prometheus.HistogramOpts{
+		Namespace: namespaceExecution,
+		Subsystem: subsystemProvider,
+		Name:      "chunk_data_pack_response_dispatch_time_ms",
+		Help:      "the total time spent on dispatching a chunk data pack response on the network milliseconds",
+		Buckets:   []float64{100, 500, 1000, 1500, 2000, 2500, 3000, 6000},
+	})
+
+	chunkDataPackQueryTime := promauto.NewHistogram(prometheus.HistogramOpts{
+		Namespace: namespaceExecution,
+		Subsystem: subsystemProvider,
+		Name:      "chunk_data_pack_query_time_ms",
+		Help:      "the total time spent on querying a chunk data pack from database",
+		Buckets:   []float64{100, 500, 1000, 1500, 2000, 2500, 3000, 6000},
```

Comment:
Suggesting to use second units here, to align with [Prometheus best practices](https://prometheus.io/docs/practices/naming/#metric-names):
```suggestion
	chunkDataPackResponseDispatchTime := promauto.NewHistogram(prometheus.HistogramOpts{
		Namespace: namespaceExecution,
		Subsystem: subsystemProvider,
		Name:      "chunk_data_pack_response_dispatch_time_seconds",
		Help:      "the total time spent on dispatching a chunk data pack response on the network in seconds",
		Buckets:   []float64{.1, .5, 1, 1.5, 2, 2.5, 3, 6},
	})

	chunkDataPackQueryTime := promauto.NewHistogram(prometheus.HistogramOpts{
		Namespace: namespaceExecution,
		Subsystem: subsystemProvider,
		Name:      "chunk_data_pack_query_time_seconds",
		Help:      "the total time spent on querying a chunk data pack from database in seconds",
		Buckets:   []float64{.01, .05, .1, .5, 1, 2, 5},
```

Also suggesting to adjust down the database query buckets, as this should typically be faster than what was the smallest bucket of 100ms.